docgen/parser/parser.go

```diff
@@ -219,14 +268,19 @@ func populateNestedMessages(message *Message, messages map[string]*Message) {
 }
 
 func parseFields(path string, comments map[string]*Comment, fields []*google_protobuf.FieldDescriptorProto) []*Field {
-	res := make([]*Field, len(fields))
+	var res []*Field
 	for i, f := range fields {
-		res[i] = &Field{
+		// check if this field is a map which should be handled by the extractMapField
+		if strings.Contains(f.GetTypeName(), f.GetName()+"Entry") {
```

Comment:
This seems a bit fragile. What if a type is really just named `FooEntry` because the user called it that?

It seems like you could use `MessageOptions.MapEntry`, which is a `*bool`.

Reply:
@mvdan this particular step checks through fields so messageoptions isn't a property of the struct. Also the line checks if the type name (`.package.Name`) is the same as (`.package.<NameAttribute>Entry`). So if the User named the field FooEntry the type name would be FooEntryEntry